types/lodash/common/common.d.ts

```diff
@@ -127,34 +127,29 @@ declare module "../index" {
          * @param key The key of the value to remove.
          * @return Returns `true` if the entry was removed successfully, else `false`.
          */
-        delete(key: string): boolean;
+        delete(key: any): boolean;
```

Comment:
Just out of curiosity, what is the motivation behind making these keys `any`?

Reply:
They are strings when the default cache key resolver is used. But if we pass an alternative resolver (the second argument of `memoize`), it can return values of any types. We just need to make sure that our cache supports keys of that type. E.g. `WeakMap` doesn't support strings and other primitives.

It's probably possible to use generics here, but this API is tricky: the cache and resolver types are tightly coupled, however the resolver is passed as an argument whereas the cache is set via setters: `_.memoize.Cache` (constructor) or `memoizedFunction.cache` (instance).